### PR TITLE
String indices

### PR DESCRIPTION
**Pull request by sdweber422**

### CODE REVIEW

File: public/javascripts/stringIndices.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  constructor(stringToParse) {
+    this._parsedString = StringIndices.parseString(stringToParse)
+  }
+
+  static parseString(stringToParse='') {
+    return stringToParse.split(' ')
+  }
+
+  getWordAtIndex(index=0) {
```
Comment (jaredatron) [inline public/javascripts/stringIndices.js]: why isn't this just
```js
getWordAtIndex(index=0) {
  return this._parsedString[index]
}
```

File: public/javascripts/stringIndices.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    return (index < 1 || index > this._parsedString.length)
+    ? ''
```
Comment (jaredatron) [inline public/javascripts/stringIndices.js]: I'd have indented this ternary.

File: routes/index.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ const express = require('express');
 const router = express.Router();
 const numberGuesser = require('../public/javascripts/NumberGuesser')
 const ScrabbleBag = require('../public/javascripts/ScrabbleBag')
+const StringIndices = require('../public/javascripts/StringIndices')
```
Comment (jaredatron) [inline routes/index.js:5]: its weird to me that these are in `../public/javascripts/`

File: routes/index.js [hunk 4]
```diff
@@ -20,4 +21,11 @@ router.get('/scrabbleBag', function(request, response, next) {
   response.render('scrabbleBag', { results: scrabbleBag.displayTiles(), errors: scrabbleBag.error() })
 })
 
+router.get('/stringIndices', function(request, response, next) {
+  const { stringToParse, index } = request.query
+  const stringEntry = new StringIndices(stringToParse)
+  const wordAtPosition = stringEntry.getWordAtIndex(index)
+  response.render('stringIndices', { stringToParse: stringToParse, returnedWord: wordAtPosition, index: index })
```
Comment (jaredatron) [inline routes/index.js]: I'd make this multiline
```js
response.render('stringIndices', {
  stringToParse: stringToParse, 
  returnedWord: wordAtPosition, 
  index: index,
})
```

File: test/StringIndices_spec.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+const StringIndices = require('../public/javascripts/StringIndices')
+const expect = require('chai').expect
+
+describe('StringIndices', function() {
+  let stringTest = new StringIndices('Please, help yourself to some')
```
Comment (jaredatron) [inline test/StringIndices_spec.js]: this is going to cause "test bleedover" you need to reinitialize this value `beforeEach` spec

```js
describe('StringIndices', function() {
  let stringTest
  beforeEach(function(){
    stringTest = new StringIndices('Please, help yourself to some')
  })
  …
})
```

File: test/StringIndices_spec.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    it('should return an array with an empty string if no string is given', function() {
+      expect(StringIndices.parseString()).to.eql([''])
+    })
+  })
+  describe('#getWordAtIndex', function() {
+    it('should return the word at word position given', function() {
+      expect(stringTest.getWordAtIndex(2)).to.eql('help')
+    })
+    it('should return an empty string for a word position below 1 or above the string length', function() {
+      expect(stringTest.getWordAtIndex(0)).to.eql('')
```
Comment (jaredatron) [inline test/StringIndices_spec.js:26]: is the index really supposed to start a 1 instead of 0?

Comment (sdweber422) [reply test/StringIndices_spec.js:26]: @deadlyicon Yes. And thanks for the wonderful feedback!

**Pull request merged**